phoenix/app/apiProducts/lib/apiProducts_web/controllers/product_controller.ex

```diff
@@ -0,0 +1,54 @@
+defmodule ApiProductsWeb.ProductController do
+  use ApiProductsWeb, :controller
+    
+  alias ApiProducts.Catalog
+  alias ApiProducts.Catalog.Product
+  
+  plug :find_product when action in [:show, :update, :delete]
```

Comment:
Para usar o módulo que tu criou para o plug no lugar da função:

`plug ApiProductsWeb.Plugs.PlugId when action in [:show, :update, :delete]`